File: agent-connect-android/app/src/main/java/sg/searchhouse/agentconnect/model/api/listing/ListingResultPO.kt
package sg.searchhouse.agentconnect.model.api.listing

import com.google.gson.annotations.SerializedName

data class ListingResultPO(
    @SerializedName("listingPOs")
    val listingPOs: List<ListingPO>,
    @SerializedName("total")
    val total: Int
)

File: srxproperty/src/dataObject/ListingResultPO.js
import { ObjectUtil } from "../utils";
import { ListingPO } from "../dataObject";

class ListingResultPO {
  constructor(data) {
    if (!ObjectUtil.isEmpty(data)) {
      this.total = data.total;
      this.type = data.type;

      // for listing
      this.listingPOs = [];
      if (!ObjectUtil.isEmpty(data.listingPOs)) {
        data.listingPOs.map(item => {
          this.listingPOs.push(new ListingPO(item));
        });
      }
    }
  }
}

export { ListingResultPO };
